slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java

```diff
@@ -305,7 +308,7 @@ private NativeTree simpleNativeTree(ParserRuleContext ctx) {
       return new NativeTreeImpl(meta(ctx), new SNativeKind(ctx, ctx.getText()), Collections.emptyList());
     }
 
-    private List<Tree> list(List<? extends ParseTree> rawChildren) {
+    private List list(List<? extends ParseTree> rawChildren) {
```

Comment:
I don't like to have such a generic list as return type...
You could cast the returned list instead in the part that you need.
`resultList.stream().map(ParameterTree.class::cast).collect(toList());`